index.js

```diff
@@ -413,6 +413,105 @@ NYC.prototype.report = function () {
   }
 }
 
+// XXX(@isaacs) Index generation should move to istanbul-lib-processinfo
+NYC.prototype.writeProcessIndex = function () {
+  const dir = this.processInfoDirectory()
+  const pidToUid = new Map()
+  const infoByUid = new Map()
+  const eidToUid = new Map()
+  const infos = fs.readdirSync(dir).filter(f => f !== 'index.json').map(f => {
+    try {
+      const info = JSON.parse(fs.readFileSync(path.resolve(dir, f), 'utf-8'))
+      // on thiis first read pass, also map the pids to uuids
+      info.uuid = path.basename(f, '.json')
+      pidToUid.set(info.uuid, info.pid)
+      pidToUid.set(info.pid, info.uuid)
+      infoByUid.set(info.uuid, info)
+      if (info.externalId) {
+        eidToUid.set(info.externalId, info.uuid)
+      }
+      return info
+    } catch (er) {
+      return null
+    }
+  }).filter(Boolean)
+
+  // create all the parent-child links and write back the updated info
+  const needsUpdate = new Set()
+  infos.forEach(info => {
+    if (info.ppid && info.ppid !== '0' && !info.parent) {
+      info.parent = pidToUid.get(info.ppid)
+      needsUpdate.add(info)
+    }
+    if (info.parent) {
+      const parentInfo = infoByUid.get(info.parent)
+      if (parentInfo.children.indexOf(info.uuid) === -1) {
+        parentInfo.children.push(info.uuid)
+        needsUpdate.add(parentInfo)
+      }
+    }
+  })
+
+  // figure out which files were touched by each process.
+  const files = infos.reduce((files, info) => {
+    if (!info.files) {
+      try {
+        info.files = Object.keys(JSON.parse(fs.readFileSync(
+          path.resolve(this.tempDirectory(), info.coverageFilename),
+          'utf-8'
+        )))
+      } catch (er) {
+        return files
+      }
+      needsUpdate.add(info)
+    }
+    info.files.forEach(f => {
+      files[f] = files[f] || []
+      files[f].push(info.uuid)
+    })
+    return files
+  }, {})
+
+  // build the actual index!
+  const index = infos.reduce((index, info) => {
+    index.processes[info.uuid] = {}
+    if (info.parent) {
+      index.processes[info.uuid].parent = info.parent
+    }
+    if (info.externalId) {
+      index.processes[info.uuid].externalId = info.externalId
+      index.externalIds[info.externalId] = {
+        root: info.uuid,
+        children: info.children
+      }
+    }
+    if (info.children && info.children.length) {
+      index.processes[info.uuid].children = Array.from(info.children)
```

Comment:
I think this can be done unconditionally (I'd prefer `index.processes[info.uuid].children` to be an empty array rather than omitted.